task22/js/main.js

```diff
@@ -1,151 +1,181 @@
     }
-    tbodyElement.appendChild(tableBodyFragment);
-    return tbodyElement;
+  }
+  tbodyElement.appendChild(tableBodyFragment);
+  return tbodyElement;
 };
 
-const getCreatedSortButton = () => {
-    const sortButton = document.createElement("button");
-    const sortImage = createElementWithClassName("img", "sort-image");
-    sortButton.id = "js-sortButton";
-    sortImage.dataset.status = "default";
-    sortButton.appendChild(sortImage);
-    return sortButton;
+const getCreatedSortButtons = (columnKey) => {
+  const sortButton = createElementWithClassName("button", "js-sortButton");
+  const sortImage = createElementWithClassName("img", "sort-image");
+  sortButton.id = `${columnKey}`;
+  sortImage.dataset.status = "default";
+  //   sortImage.classList.add = "js-sortImage";
+  sortImage.src = "../img/both.svg";
+  sortImage.alt = "default-image";
+  sortButton.appendChild(sortImage);
+  return sortButton;
 };
 
-const addEventForSortButton = (userContentsData) => {
-    const sortButton = document.getElementById("js-sortButton");
-    sortButton.addEventListener('click', (e) => {
-        const nextStatus = changeSortStatus(e.target.dataset.status);
-        e.target.dataset.status = nextStatus;
-
-        sortContents(userContentsData, nextStatus);        
-    })
-}
-
 const changeSortStatus = (status) => {
-    switch(status){
-        case "default":
-            return "asc";
-        case "asc":
-            return "desc";
-        case "desc":
-        default:
-            return "default";
+  switch (status) {
+    case "default":
+      return "asc";
+    case "asc":
+      return "desc";
+    case "desc":
+    default:
+      return "default";
+  }
+};
+
+const notTargetIsDefault = (target, nextStatus) => {
+  const notTarget = [...document.querySelectorAll(".sort-image")].filter(
+    (column) => column !== target
+  );
+  notTarget.forEach((column) => {
+    if (nextStatus !== "default") {
+      column.dataset.status = "default";
     }
-}
+  });
+};
+
+const addEventForSortButton = (userContentsData) => {
+  const sortButtons = document.querySelectorAll(".js-sortButton");
+  sortButtons.forEach((button) => {
+    button.addEventListener("click", (e) => {
+      const nextStatus = changeSortStatus(e.target.dataset.status);
+      e.target.dataset.status = nextStatus;
+      const targetId = e.currentTarget.id;
+      sortContents(targetId, userContentsData, nextStatus);
+      notTargetIsDefault(e.target, nextStatus);
+    });
+  });
+};
 
 const renderTableData = (userContentsData) => {
-    document.getElementById("js-tableBody").remove();
-    const table = document.querySelector("table");
-    table.appendChild(getCreatedTableBody(userContentsData));
+  document.getElementById("js-tableBody").remove();
+  const table = document.querySelector("table");
+  table.appendChild(getCreatedTableBody(userContentsData));
+};
+
+const sortContents = (targetColum, userContentsData, status) => {
```

Comment:
typo
```suggestion
const sortContents = (targetColumn, userContentsData, status) => {
```

Reply:
Thank you for pointing this out. My apologies for the typo. It has been corrected.
[126c1ad](https://github.com/seeeen044/JavaScript_handson/pull/24/commits/126c1ad5a8ff2c17ed83fd589653e8b9f6c5bcb5)